drivers/serial/uart_renesas_ra8_sci_b.c

```diff
@@ -0,0 +1,1165 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra8_uart_sci_b
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+#include "r_sci_b_uart.h"
+#include "r_dtc.h"
+#include "r_transfer_api.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ra8_uart_sci);
+
+#if defined(CONFIG_UART_ASYNC_API)
+void sci_b_uart_rxi_isr(void);
+void sci_b_uart_txi_isr(void);
+void sci_b_uart_tei_isr(void);
+void sci_b_uart_eri_isr(void);
+#endif
+
+struct uart_ra_sci_b_config {
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct uart_ra_sci_b_data {
+	const struct device *dev;
```

Comment:
`struct device` is usually placed on config structure.

Reply:
We need it to handle for Asynchronous case. We saw that some others have similar implementation (ex. drivers\serial\uart_stm32 and drivers\serial\uart_ns16550)